Source/WarpX.cpp

```diff
@@ -1176,52 +1178,57 @@ WarpX::ReadParameters ()
         // Check whether the default comoving velocity should be used
         bool use_default_v_comoving = false;
         pp_psatd.query("use_default_v_comoving", use_default_v_comoving);
+
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+            !use_default_v_comoving || boosted_frame,
+            "psatd.use_default_v_comoving = 1 can be used only if warpx.gamma_boost is also set"
+        );
+
         if (use_default_v_comoving == true && boosted_frame == true)
         {
             m_v_comoving[2] = -std::sqrt(1._rt - 1._rt / (gamma_boost * gamma_boost));
         }
-        else if (use_default_v_comoving == true && boosted_frame == false)
-        {
-            amrex::Abort("psatd.use_default_v_comoving = 1 can be used only if warpx.gamma_boost is also set");
-        }
         else
         {
             queryArrWithParser(pp_psatd, "v_comoving", m_v_comoving, 0, 3);
         }
 
-        // Galilean and comoving algorithms should not be used together
-        if (m_v_galilean[0] != 0. || m_v_galilean[1] != 0. || m_v_galilean[2] != 0.)
-        {
-            if (m_v_comoving[0] != 0. || m_v_comoving[1] != 0. || m_v_comoving[2] != 0.)
-            {
-                amrex::Abort("Galilean and comoving algorithms should not be used together");
-            }
-        }
-
         // Scale the Galilean/comoving velocity by the speed of light
-        for (int i=0; i<3; i++) m_v_galilean[i] *= PhysConst::c;
-        for (int i=0; i<3; i++) m_v_comoving[i] *= PhysConst::c;
+        for (auto& vv : m_v_galilean) vv*= PhysConst::c;
+        for (auto& vv : m_v_comoving) vv*= PhysConst::c;
+
+        const auto v_galilean_is_zero =
+            std::all_of(m_v_galilean.begin(), m_v_galilean.end(),
+            [](const auto& val){return val == 0.;});
+
+        const auto v_comoving_is_zero =
+            std::all_of(m_v_comoving.begin(), m_v_comoving.end(),
+            [](const auto& val){return val == 0.;});
+
+
+        // Galilean and comoving algorithms should not be used together
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+            v_galilean_is_zero || v_comoving_is_zero,
```

Comment:
According to the message, I think this needs to be an `xor` not an `or`.
```suggestion
            v_galilean_is_zero ^ v_comoving_is_zero,
```

Reply:
Uhm... I am not actually sure about this... the nested `if` in the original code is equivalent to:
`if ( !v_galilean_is_zero && !v_comoving_is_zero)`, so the negation of that condition should be:
`v_galilean_is_zero || v_comoving_is_zero` . Moreover, if they are both zero, it means that you are not using neither the galilean algorithm nor the comoving algorithm, which should be fine if I am not mistaken...

Reply:
You are right, I must have overlooked the change to `_is_zero` variables last night here.